packages/ddp-server/livedata_server.js

```diff
@@ -1644,20 +1648,29 @@ _.extend(Server.prototype, {
     };
     var connection = null;
     var currentInvocation = DDP._CurrentInvocation.get();
+    var currentPublicationInvocation = DDP._CurrentPublicationInvocation.get();
+    var randomSeed = null;
     if (currentInvocation) {
       userId = currentInvocation.userId;
       setUserId = function(userId) {
         currentInvocation.setUserId(userId);
       };
       connection = currentInvocation.connection;
+      randomSeed = DDPCommon.makeRpcSeed(currentInvocation, name);
+    } else if (currentPublicationInvocation) {
+      userId = currentPublicationInvocation.userId;
+      setUserId = function(userId) {
```

Comment:
You are right. You are doing the same as what initial method invocation is doing.